File: Objective-C/ExampleApplication/ExampleApplication/TableViewController.m
//
//  ViewController.m
//  ExampleApplication
//
//  Created by Levent ORAL on 22.11.2018.
//  Copyright © 2018 App Samurai Inc. All rights reserved.
//

#import "TableViewController.h"

@interface TableViewController ()
    @property(nonatomic, copy) NSArray *AdFormatList;
    
    @property(nonatomic, copy) NSArray *SegueList;
@end

@implementation TableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.AdFormatList = @[@"Banner 320x50 from Storyboard",
                          @"Banner 320x50 Programatically",
                          @"Interstitial",
                          @"Rewarded Video"];
    self.SegueList = @[@"banner350x50FromStoryboardSegue",
                      @"banner320x50ProgramaticallySegue",
                      @"interstitialSegue",
                      @"rewardBasedVideoAdSegue"];
}

#pragma mark - Table view data source
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.AdFormatList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *basicCell = [tableView dequeueReusableCellWithIdentifier:@"BasicCell"];
    basicCell.textLabel.text = self.AdFormatList[indexPath.row];
    return basicCell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    [self performSegueWithIdentifier:self.SegueList[indexPath.row] sender:self];
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}
@end

File: Swift/ExampleApplication/ExampleApplication/TableViewController.swift
//
//  ViewController.swift
//  ExampleApplication
//
//  Created by Levent ORAL on 22.11.2018.
//  Copyright © 2018 App Samurai Inc. All rights reserved.
//

import UIKit

class TableViewController: UITableViewController {
    
    private let AdFormatList = [
        ("Banner 320x50 from Storyboard", "banner350x50FromStoryboardSegue"),
        ("Banner 320x50 Programatically", "banner320x50ProgramaticallySegue"),
        ("Interstitial", "interstitialSegue"),
        ("Rewarded Video", "rewardBasedVideoAdSegue")
    ]

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
     // MARK: - Table view data source
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return AdFormatList.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let basicCell = tableView.dequeueReusableCell(withIdentifier: "BasicCell")!
        basicCell.textLabel?.text = AdFormatList[indexPath.row].0
        return basicCell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        performSegue(withIdentifier: AdFormatList[indexPath.row].1, sender: self)
    }
}

